src/Microsoft.DotNet.Build.Tasks/PackageFiles/tests.targets

```diff
@@ -150,20 +151,19 @@
   <Target Name="GenerateTestBindingRedirects"
           Condition="'$(TestAgainstDesktop)' == 'true' and '$(LocalAppConfigFile)' == ''"
           DependsOnTargets="DiscoverTestInputs;DiscoverTestDependencies">
-    
     <ItemGroup>
-      <_CombinedListOfTestAssets Include="@(IncludedFileForRunnerScript->'%(SourcePath)')" />
-      <_CombinedListOfTestAssets Include="@(SourcesToCopyToTestDir)" />
-      
-      <_FilteredAssembliesForBindingRedirect Include="@(_CombinedListOfTestAssets)" Condition="'%(Extension)'=='.dll' or '%(Extension)'=='.exe'" />
-      <_ExecutablesForBindingRedirect Include="@(SourcesToCopyToTestDir)" Condition="'%(Extension)'!='.pdb'" />
+      <_FilteredAssembliesForBindingRedirect Include="%(_FinalListOfTestAssetsByFileName.SourcePath)" Condition="'%(_FinalListOfTestAssetsByFileName.Extension)'=='.dll' or '%(_FinalListOfTestAssetsByFileName.Extension)'=='.exe'" />
+      <ExecutablesForBindingRedirect Include="%(_SourcesToCopyToTestDirByFileName.SourcePath)" Condition="'%(_SourcesToCopyToTestDirByFileName.Extension)'!='.pdb'"/>
     </ItemGroup>
-    
-    <GetAssemblyIdentity AssemblyFiles="@(_FilteredAssembliesForBindingRedirect)">
-      <Output TaskParameter="Assemblies" ItemName="TestAssemblyIdentities" />
+    <FilterNativeAssemblies Condition="'$(TestAgainstDesktop)' == 'true'" Assemblies="@(_FilteredAssembliesForBindingRedirect)">
+      <Output TaskParameter="ManagedAssemblies" ItemName="ManagedTestAssemblyIdentities"/>
+    </FilterNativeAssemblies>
+    <GetAssemblyIdentity Condition="'$(TestAgainstDesktop)' == 'true'" AssemblyFiles="@(ManagedTestAssemblyIdentities)">
+      <Output TaskParameter="Assemblies" ItemName="TestAssemblyIdentities"/>
     </GetAssemblyIdentity>
-
-    <GenerateBindingRedirect Executables="@(_ExecutablesForBindingRedirect)"
+    
+    <GenerateBindingRedirect Condition="'$(TestAgainstDesktop)' == 'true' and '$(LocalAppConfigFile)' == ''"
```

Comment:
This change isn't needed because the Condition is in the `GenerateTestBindingRedirects` target.  The Condition is also not needed for the call to the `FilterNativeAssemblies` task and the `GetAssemblyIdentity` task in this target.